### DISCUSSION TITLE
x86-32  interrupts for Assembly game development.

### DISCUSSION THREAD

**Discussion by ghost** [community/community#121377] [Programming Help]
### Body

* 3D Graphics,
* Audio,
* Physics

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by CopperEagle** [comment]
[Here](https://www.ctyme.com/intr/int.htm) is a table of all interrupts, for many platforms (Linux, FreeBSD, etc.).

Developing a 3D game in Assembly will be time consuming. For context: [I wrote a](https://github.com/CopperEagle/Visualplane) 2D animation engine in C++ using OpenGL.  2D, no physics. It had an audio spectrum analyzer feature that I ultimately didn't upload. It took a total of several weeks nonstop development. 

If you want to program a high performance game, check out [OpenGL](https://www.opengl-tutorial.org/) for C/C++. It's a graphics library, allowing you to render on the GPU. It's what I many games used albeit it is slowly being replaced with Vulkan. I should mention [SDL](https://www.libsdl.org/) for audio, window, mouse, etc.

If you like Python, check out [Pygame](https://pypi.org/project/pygame/).
